### PR TITLE
Fix for `ft-transfer` mismatching arguments

### PR DESCRIPTION
**Pull request by adizere**
Closes: #869

## Description

See the issue for a discussion of the scenario for this bug.

______

For contributor use:

- [X] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer-cli/src/commands/tx/transfer.rs [hunk 1]
```diff
@@ -120,6 +121,49 @@ impl Runnable for TxIcs20MsgTransferCmd {
...
+        );
+        if src_chain_client_state.chain_id != self.dst_chain_id {
+            return Output::error(
+                format!("The requested port/channel ({}/{}) provides a path from chain '{}' to \
+                 chain '{}' (not to the destination chain '{}'). Bailing due to mismatching arguments.",
+                        opts.packet_src_port_id, opts.packet_src_channel_id,
+                        self.src_chain_id,
+                        src_chain_client_state.chain_id, self.dst_chain_id)).exit();
+        }
+
```
Comment (ancazamfir) [inline relayer-cli/src/commands/tx/transfer.rs:165]: Should we just remove `dst_chain_id`?

**Review by ancazamfir — APPROVED** [review]
Looks good, thanks @Adi. Just one comment if we should remove the dst chain parameter. 
Or we leave it to a general PR that cleans all redundant params in `raw tx` commands.

File: relayer-cli/src/commands/tx/transfer.rs [hunk 2]
```diff
@@ -120,6 +121,49 @@ impl Runnable for TxIcs20MsgTransferCmd {
...
+                &opts.packet_src_port_id,
+                &opts.packet_src_channel_id,
+                Height::zero(),
+            )
+            .unwrap_or_else(exit_with_unrecoverable_error);
+        // TODO: Support for multi-hop channels will impact this.
+        let conn_id = match channel_end.connection_hops.first() {
+            None => {
+                return Output::error(format!(
+                    "Could not retrieve the connection hop underlying channel end {:?} on chain {}",
```
Comment (ancazamfir) [inline relayer-cli/src/commands/tx/transfer.rs:139]:
```suggestion
                    "could not retrieve the connection hop underlying channel end {:?} on chain {}",
```
Also in other places. I prefer errors and logs to start lower case. Up to you for this one.

File: relayer-cli/src/commands/tx/transfer.rs (hunk 1 above)
Comment (adizere) [reply relayer-cli/src/commands/tx/transfer.rs:165]: I don't think we should remove the option `dst_chain_id`.
I can see some value in it: Given that the end-user is forced to specify the name of the destination chain where they are sending the tokens, I think this makes it a good failsafe mechanism.

File: relayer-cli/src/commands/tx/transfer.rs (hunk 2 above)
Comment (adizere) [reply relayer-cli/src/commands/tx/transfer.rs]: done for the log messages included in this PR! e7944169

ACK, will keep in mind the lower-case approach for future.

**Pull request merged**